assets/queries/k8s/service_with_external_load_balance/query.rego

```diff
@@ -1,34 +1,52 @@
 package Cx
 
 CxPolicy[result] {
-	service := input.document[i]
-	service.kind == "Service"
-	metadata := service.metadata
+	document := input.document[i]
+	object.get(document, "kind", "undefined") == "Service"
 
-	not CheckIfDestinationPortExists(service)
+    metadata = document.metadata
+	document.spec.type == "LoadBalancer"
+
+	object.get(metadata, "annotations", "undefined") == "undefined"
 
 	result := {
 		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name=%s.spec", [metadata.name]),
+		"searchKey": sprintf("metadata.name={{%s}}", [metadata.name]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("metadata.name=%s is not exposing a workload", [metadata.name]),
-		"keyActualValue": sprintf("metadata.name=%s is exposing a workload", [metadata.name]),
+		"keyExpectedValue": "'metadata.annotations' is set",
+		"keyActualValue": "'metadata.annotations' is undefined",
 	}
 }
 
-CheckIfDestinationPortExists(service) = result {
-	documents := input.document
-	pod := [pod | documents[index].spec.template.metadata.labels.app == service.spec.selector.app; pod = documents[index]]
-	service.spec.type == "LoadBalancer"
+CxPolicy[result] {
+	document := input.document[i]
+	object.get(document, "kind", "undefined") == "Service"
+
+    metadata = document.metadata
+	document.spec.type == "LoadBalancer"
+
+	object.get(metadata, "annotations", "undefined") != "undefined"
+
+    annotations = metadata.annotations
+    not checkLoadBalancer(annotations)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("metadata.name={{%s}}.annotations", [metadata.name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("metadata.name={{%s}} using an external Load Balancer provider by cloud provider", [metadata.name]),
+		"keyActualValue": sprintf("metadata.name={{%s}} is exposing a workload, not using an external Load Balancer provider by cloud provider", [metadata.name]),
+	}
+}
 
-	result := contains(pod, service.spec.selector.app)
-	contains(pod, service.spec.ports[k].targetPort)
+checkLoadBalancer(annotation) {
+    annotation["cloud.google.com/load-balancer-type"] == "Internal"
```

Comment:
This is a check for older versions for GKE, as you can see [here](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing)

I think we should also check this:
`cloud.google.com/load-balancer-type: "Internal"`